### DISCUSSION TITLE
Which packages currently work?

### DISCUSSION THREAD

**Discussion by ghost** [pulsar-edit/.github#129] [Community Packages]
Which all packages work with Pulsar?

If anyone has any ideas, I'm currently looking for packages to transform Pulsar Edit into a function Python IDE.

I mainly need something like data-tips/documentation and a package that can highlight/underline/show errors in my Python files/code.

**Comment by confused-Techie** [comment]
I'd honestly say the majority of packages work just fine.

It would be more reasonable to compile a list of packages that don't work.

Which what I know of:

* [hydrogen](https://web.pulsar-edit.dev/packages/hydrogen) - Has been fixed on GitHub, the fixed version is not available to install from the Pulsar Settings, but can be installed manually from their GitHub
* [tron-green-ui](https://web.pulsar-edit.dev/packages/tron-green-ui) - Mostly works fine, but has some minor fixes when installing via GitHub

The ones above I have personally checked, beyond that you can look at open issues on [pulsar](https://github.com/pulsar-edit/pulsar) to see packages community members are having issues with.

We do plan in the future on having a system to add notices to packages that have issues, but that has not currently been implemented.

**Reply by ghost** [reply]
Interesting… I’ve been having problems finding a working Python linter and package to find errors in code. linter-pylama doesn’t work at all. Neither does the linter base or linter-default-ui. 

I’ll have to stick with VSCode for now, seems like Atom was broken before the sunset. And for now, replacements are under development. Though Pulsar looks to have a bright future.

**Reply by savetheclocktower** [reply]
I've found `linter` to be working just fine, and `linter-default-ui` to be mostly working albeit with some annoying bugs. But those bugs were present in Atom as well. And, yes, they're due for replacements anyway.

**Reply by ghost** [reply]
I'll try them out again... Thanks for pointing that out! :)

**Comment by ghost** [comment]
There is also a lack of built-in language support packages like linters and debuggers and error-finders(?). I’m assuming this is because Atom and Pulsar are “text editors” rather than code editors or IDEs.

**Reply by confused-Techie** [reply]
Yeah the majority of core packages focus on things like syntax highlighting, and managing the text of the editor, such as wrapping text, and such.

Although there are some core `ide`s for languages, that mostly do things like autocompletion, the actual error finding never seemed to be a huge focus of Atom's team, so isn't something we have a lot of ourselves now, but of course that may change in the future, or may not.

**Reply by ghost** [reply]
Okay, thank you for your response and time! :D
